### PR TITLE
first version of all tests using the spring-boot and mock-mvc (remains

### PR DESCRIPTION
**Pull request by SrdjanStevanetic**
to set-up solr and mongo in docker)

### CODE REVIEW

File: annotation-web/src/main/java/eu/europeana/annotation/web/service/controller/admin/WhitelistRest.java [hunk 1]
```diff
@@ -128,7 +128,7 @@ public ResponseEntity<String> createWhitelistEntry(@RequestBody String whitelist
 	return buildResponse(jsonStr);
     }
 
-    @RequestMapping(value = "/load", method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_VALUE)
+    @RequestMapping(value = "/load", method = RequestMethod.GET)
```
Comment (gsergiu) [inline annotation-web/src/main/java/eu/europeana/annotation/web/service/controller/admin/WhitelistRest.java:131]: the produces are needed for content negotiation mechanism

File: annotation-web/src/main/java/eu/europeana/annotation/web/service/controller/admin/WhitelistRest.java [hunk 2]
```diff
@@ -150,7 +150,7 @@ public ResponseEntity<String> loadDefaultWhitelist(
     }
 
     @DELETE
-    @RequestMapping(value = "/deleteall", method = RequestMethod.DELETE, produces = MediaType.APPLICATION_JSON_VALUE)
+    @RequestMapping(value = "/deleteall", method = RequestMethod.DELETE)
```
Comment (gsergiu) [inline annotation-web/src/main/java/eu/europeana/annotation/web/service/controller/admin/WhitelistRest.java:153]: the produces are needed for content negotiation mechanism

File: annotation-web/src/main/java/eu/europeana/annotation/web/service/controller/jsonld/WebAnnotationAuxiliaryMethodsRest.java [hunk 3]
```diff
@@ -66,10 +66,9 @@ public ResponseEntity<String> createAnnotationByTypeJsonld(
 	return storeAnnotation(motivation, indexOnCreate, annotation, authentication);
     }
 
-    @RequestMapping(value = "/annotations/deleted", method = RequestMethod.GET, produces = {
-	    HttpHeaders.CONTENT_TYPE_JSON_UTF8 })
+    @RequestMapping(value = "/annotations/deleted", method = RequestMethod.GET)
```
Comment (gsergiu) [inline annotation-web/src/main/java/eu/europeana/annotation/web/service/controller/jsonld/WebAnnotationAuxiliaryMethodsRest.java:69]: the produces are needed for content negotiation mechanism and for filling the content type response header

**Review by gsergiu — CHANGES_REQUESTED** [review]

File: annotation-web/pom.xml [hunk 4]
```diff
@@ -231,10 +231,20 @@
...
+<!--         <dependency> -->
+<!--             <groupId>dev.morphia.morphia</groupId> -->
+<!--             <artifactId>morphia-core</artifactId> -->
+<!--             <version>${version.morphia}</version> -->
+<!--         </dependency> -->
+		<!-- https://mvnrepository.com/artifact/org.mongodb/mongo-java-driver -->
+		<dependency>
+		    <groupId>org.mongodb</groupId>
+		    <artifactId>mongo-java-driver</artifactId>
+		    <version>3.12.11</version>
```
[the remark this answers is not in the document]
Comment (SrdjanStevanetic) [reply annotation-web/pom.xml:243]: Yes it is mandatory, please see the comment in the MongoConfig class.

File: annotation-web/src/main/java/eu/europeana/annotation/web/service/controller/admin/WhitelistRest.java (hunk 2 above)
Comment (SrdjanStevanetic) [reply annotation-web/src/main/java/eu/europeana/annotation/web/service/controller/admin/WhitelistRest.java:153]: Ok.

File: annotation-web/src/main/java/eu/europeana/annotation/web/service/controller/jsonld/WebAnnotationAuxiliaryMethodsRest.java (hunk 3 above)
Comment (SrdjanStevanetic) [reply annotation-web/src/main/java/eu/europeana/annotation/web/service/controller/jsonld/WebAnnotationAuxiliaryMethodsRest.java:69]: Ok.

**Pull request merged**